File: leetcode/0058-length-last-word/0058-solution.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i = len(s) - 1

        # Skip over trailing whitespace to find the end of the word
        while i > 0 and s[i] == " ":
            i -= 1

        # Save where the word ends
        end = i

        # Count non-whitespace characters
        while i >= 0 and s[i] != " ":
            i -= 1

        return end - i

File: leetcode/0058-length-last-word/0058-solution.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        
        int i{ static_cast<int>(s.size()) - 1 };

        while (i > 0 && s[i] == ' ')
        {
            i--;
        }

        int end{i};

        while (i >= 0 &&  s[i] != ' ')
        {
            i--;
        }

        return end - i;
    }
};
